### DISCUSSION TITLE
New in v1.3.0 - ✨Inline Transformations

### DISCUSSION THREAD

**Discussion by olimorris** [olimorris/codecompanion.nvim#130] [Announcements]
As outlined in #112, [Zed AI](https://zed.dev/blog/zed-ai) has this nifty ability of being able to take code that's been generated in a chat, and add it to a buffer, via a prompt. The Zed team call it _Inline Transformations_ and I like it very much.

So as of v1.3.0, I've added Inline Transformations into the plugin:

https://github.com/user-attachments/assets/20e96413-16f6-4764-ada8-0fd0ffe8f77f

When you run an inline transformation with `:CodeCompanion <your prompt>`, the inline strategy will automatically grab the content from the last chat buffer you conversed with and the last message sent by the LLM.

I expect there may be some initial bugs so please report back under this message.

**Comment by towry** [comment]
Will there be Cursor tab AI experience?

**Reply by olimorris** [reply]
I don't know what that is. But if you'd like to share your ideas, I'm open.

**Reply by olimorris** [reply]
If you're referring to code completion: https://github.com/olimorris/codecompanion.nvim/discussions/352#discussioncomment-11434001
